f5_openstack_agent/lbaasv2/drivers/bigip/monitor_manager.py

```diff
@@ -497,52 +504,63 @@ def report_device_status_task(self):
+                members = device_info['members']
+
+                for m in members:
+                    member_id = m['id']
+                    LOG.debug(member_id)
+                    LOG.debug("operating_status: %s" % m['operating_status'])
+
+                    member = {
+                        "member": {"operating_status": {}}
                     }
-                }
 
-                for bip in bigips:
+                    username = decrypt_data(m['device_info']['serial_number'],
+                                            m['device_info']['username'])
+                    password = decrypt_data(m['device_info']['serial_number'],
+                                            m['device_info']['password'])
+
                     try:
-                        hostname = bip['hostname']
-                        bigip = ManagementRoot(bip['hostname'],
-                                               bip['username'],
-                                               bip['password'],
-                                               port=bip['port'],
+                        timestamp = datetime.datetime.utcnow().strftime(
+                            '%Y-%m-%dT%H:%M:%S') + 'Z'
+                        hostname = m['mgmt_ipv4']
+                        bigip = ManagementRoot(hostname,
+                                               username,
+                                               password,
+                                               port=m['device_info']['port'],
                                                token=True)
-                        device['device']['operating_status'][hostname] = \
-                            'ONLINE'
+
+                        failover_state = bigip.tm.sys.dbs.db.load(
+                            name='failover.state'
+                        )
+                        member = {
+                            "member": {
+                                "operating_status": {
+                                    "status": "ONLINE",
+                                    "timestamp": timestamp,
+                                    "failover_state": failover_state.value,
+                                    "error": ""
+                                }
+                            }
+                        }
+
                     except Exception as e:
                         LOG.error('seems bigip %s has problem' % hostname)
-                        device['device']['operating_status'][hostname] = \
+                        member['member']['operating_status']['error'] = \
                             str(e)[:80]
-                        device['device']['operating_status']['status'] = \
-                            'SOMETHING_WRONG'
+                        member['member']['operating_status']['status'] = \
+                            'OFFLINE'
+                        member['member']['operating_status'][
+                            'failover_state'] = 'UNKNOWN'
```

Comment:
I saw that fail over state 'active' and 'standby' are low case, please also input 'unknown' to make them consistent.